### PR TITLE
[quant][graphmode] Test weight observer for dynamic quant

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39687 [quant][graphmode] Test weight observer for dynamic quant**

Summary:
Run the observer on the weight values and compare it with the calculated attributes in the graph

Test Plan:
python test/test_quantization.py test_dynamic_weight_observer

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21961907](https://our.internmc.facebook.com/intern/diff/D21961907)

### CODE REVIEW

File: test/quantization/test_quantize_script.py [hunk 1]
```diff
@@ -2559,13 +2559,51 @@ def forward(self, x):
...
         FileCheck().check_count("quantized::linear_dynamic(", 2, exactly=True) \
                    .check_not("aten::_choose_qparams_per_tensor") \
                    .run(m1.graph)
+
+        # Check to make sure weight observers run correctly
+        ref_qparams = []
+        qconfig = script_qconfig(default_dynamic_qconfig)
+        for wt in [model.res1.weight, model.res2.weight]:
+            get_forward(qconfig.weight)(wt)
+            qparams = qconfig.weight._get_method('calculate_qparams')()
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:2571]: I think you can call them directly, e.g. `qconfig.weight(wt)` and `qconfig.weight.calculate_qparams`

File: test/quantization/test_quantize_script.py [hunk 2]
```diff
@@ -2559,13 +2559,51 @@ def forward(self, x):
...
+                x = self.fc(x)
+                return self.fc2(x)
+
+        qconfig_dict = {'': default_dynamic_qconfig}
+        model = torch.jit.script(M()).eval()
+        qconfig = script_qconfig(default_dynamic_qconfig)
+        ref_qparams = []
+        for wt in [model.fc.weight, model.fc2.weight]:
+            get_forward(qconfig.weight)(wt)
+            qparams = qconfig.weight._get_method('calculate_qparams')()
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:2601]: same here

**Review by jerryzh168 — APPROVED** [review]
LG, see inline comment

File: test/quantization/test_quantize_script.py (hunk 1 above)
Comment (supriyar) [reply test/quantization/test_quantize_script.py:2571]: It doesn't seem to be directly callable `TypeError: 'torch._C.ScriptModule' object is not callable`

Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:2571]: I see, maybe you can wrap it into a RecursiveScriptModule with `wrap_cpp_module` first

**Pull request closed without merging**